bench/bench_kotlin/src/main/kotlin/math/Repr.kt

```diff
@@ -0,0 +1,40 @@
+package math
+
+import java.math.BigInteger
+
+interface Repr<t> {
+
+    fun abs(a: t): t
+
+    fun inverse(a: t): t
+
+    fun add(a: t): (t) -> t
+
+    fun sub(a: t): (t) -> t
+
+    fun lt(a: t): (t) -> Boolean
+
+    fun lte(a: t): (t) -> Boolean
+
+    fun gt(a: t): (t) -> Boolean = { b -> lt(b)(a) }
```

Comment:
I don't know much about Kotlin, but seems like there is no support for currying, should this be written in un-curried style (tupled params `lt(a, b)`) ?

Reply:
Of course. And at the same time, I can check whether the inlining can be specified.